File: CS/E5198/CustomFontsHelper.cs
using System;
using System.Drawing;
using System.Drawing.Text;
using System.Linq;
using System.Web;

namespace E5198 {
    public static class CustomFontsHelper {

        static PrivateFontCollection fontCollection;
        public static FontCollection FontCollection {
            get {
                if (fontCollection == null) {
                    fontCollection = new PrivateFontCollection();
                    fontCollection.AddFontFile(HttpContext.Current.Server.MapPath("~/Fonts/MissFajardose-Regular.ttf"));
                }
                return fontCollection;
            }
        }

        public static FontFamily GetFamily(string familyName) {
            return FontCollection.Families.FirstOrDefault(ff => ff.Name == familyName);
        }
    }
}

File: VB/E5198/CustomFontsHelper.vb
Imports System.Drawing
Imports System.Drawing.Text
Imports System.Linq
Imports System.Web

Namespace E5198

    Public Module CustomFontsHelper

        Private fontCollectionField As PrivateFontCollection

        Public ReadOnly Property FontCollection As FontCollection
            Get
                If fontCollectionField Is Nothing Then
                    fontCollectionField = New PrivateFontCollection()
                    fontCollectionField.AddFontFile(HttpContext.Current.Server.MapPath("~/Fonts/MissFajardose-Regular.ttf"))
                End If

                Return fontCollectionField
            End Get
        End Property

        Public Function GetFamily(ByVal familyName As String) As FontFamily
            Return FontCollection.Families.FirstOrDefault(Function(ff) Equals(ff.Name, familyName))
        End Function
    End Module
End Namespace
